consensus/consensus_leader.go

```diff
@@ -31,13 +31,17 @@ var (
 
 // WaitForNewBlock waits for the next new block to run consensus on
 func (consensus *Consensus) WaitForNewBlock(blockChannel chan *types.Block, stopChan chan struct{}, stoppedChan chan struct{}, startChannel chan struct{}) {
+	first := true
 	go func() {
 		defer close(stoppedChan)
 		for {
 			select {
 			default:
-				// got the signal to start consensus
-				_ = <-startChannel
+				if first && startChannel != nil {
```

Comment:
consolidate this when first = true and startChannel == nil

if you expect startChannel should be not nil then log error. Otherwise

Reply:
startChannel != nil is for protection of program stuck.